ignite/internal/tools/gen-mig-docs/main.go

```diff
@@ -0,0 +1,630 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/hexops/gotextdiff"
+	"github.com/hexops/gotextdiff/myers"
+	"github.com/hexops/gotextdiff/span"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/exec"
+	"github.com/ignite/cli/ignite/pkg/cmdrunner/step"
+	"github.com/pkg/errors"
+)
+
+const (
+	igniteCliRepository = "http://github.com/ignite/cli.git"
+	igniteBinaryPath    = "dist/ignite"
+	igniteRepoPath      = "src/github.com/ignite/cli"
+)
+
+var scaffoldCommands = map[string][]string{
+	"chain":  {"chain example --no-module"},
+	"module": {"chain example"},
+	"list": {
+		"chain example",
+		"list list1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"map": {
+		"chain example",
+		"map map1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"single": {
+		"chain example",
+		"single single1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"type": {
+		"chain example",
+		"type type1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"message": {
+		"chain example",
+		"message message1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+	"query": {
+		"chain example",
+		"query query1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints --module example --yes",
+	},
+	"packet": {
+		"chain example --no-module",
+		"module example --ibc",
+		"packet packet1 f1:string f2:strings f3:bool f4:int f5:ints f6:uint f7:uints f8:coin f9:coins --module example --yes",
+	},
+}
+
+var diffExceptions = []string{
+	"*.md",
+	"go.sum",
+	"*_test.go",
+	"*.pb.go",
+	"*.pb.gw.go",
+	"openapi.yml", // FIXME: For some reason removing this file from exceptions causes too much memory usage and OS kills the process
+	".gitignore",
+	"*.html",
+	"*.css",
+	"*.js",
+	"*.ts",
+}
+
+func main() {
+	var logger = log.New(os.Stdout, "", log.LstdFlags)
+
+	fromFlag := flag.String("from", "", "Semver tag to generate migration document from")
```

Comment:
why not use cobra?

Reply:
and maybe these should be args and not flags

Reply:
We have to use flags since both `from` and `to` could be omitted and we have a strategy for choosing default values for both of them. If we use args we can't determine which one is omitted.